lightning/src/ln/msgs.rs

```diff
@@ -2911,9 +2908,19 @@ impl<NS: Deref> ReadableArgs<(Option<PublicKey>, NS)> for InboundOnionPayload wh
 						next_blinding_override,
 					})
 				},
-				ChaChaPolyReadAdapter { readable: BlindedPaymentTlvs::Receive(ReceiveTlvs {
-					payment_secret, payment_constraints, payment_context, authentication,
-				})} => {
+				ChaChaPolyReadAdapter { readable: BlindedPaymentTlvs::Receive(mut receive_tlvs) } => {
+					if let Some((hmac, nonce)) = receive_tlvs.authentication.take() {
+						let expanded_key = node_signer.get_inbound_payment_key();
+						if receive_tlvs.verify_for_offer_payment(hmac, nonce, &expanded_key).is_err() {
+							return Err(DecodeError::InvalidValue);
+						}
+					} else {
+						return Err(DecodeError::InvalidValue);
```

Comment:
Done.

Reply:
I think we'll also fail backwards without blinding the error. That might give a hint that we're the recipient if combined with other information, not sure if also worth noting. I guess this would also mean that someone can create a blinded path terminating at us and force us to reveal that we are the last node in their inauthentic path 🤔 weird case though

Reply:
> I think we'll also fail backwards without blinding the error. That might give a hint that we're the recipient if combined with other information, not sure if also worth noting.

Would we need to use the original approach to prevent this?

> I guess this would also mean that someone can create a blinded path terminating at us and force us to reveal that we are the last node in their inauthentic path 🤔 weird case though

Not sure I understand this scenario. Don't they know we are the recipient? Or is it being revealed to someone else along the path?

Reply:
> Would we need to use the original approach to prevent this?

The original approach should prevent this, or alternately add a new `DecodeError` or somehow bubble up from the `Read` method that the error needs to be blinded. 

> Not sure I understand this scenario. Don't they know we are the recipient? Or is it being revealed to someone else along the path?

Ah, it would be potentially revealed to the hop preceding us. I'm not sure this is worth addressing since it's a fairly narrow pre-0.1 case, but yeah just wanted to point it out.